src/main/java/gui/DashboardWindow.java

```diff
@@ -0,0 +1,307 @@
+    private boolean isAnyCustomPhilosophy = true;
+
+
+    public DashboardWindow() {
+
+        factory = BusinessLogic.getFactory();
+
+        setSize(880, 800);
+        setResizable(false);
+        setTitle("Moral dilemma detector");
+        setLayout(null);
+
+
+        jButtonLoadFromFile = new JButton("Load scenario from file");
+        jButtonLoadFromFile.setBounds(20, 10, 400, 30);
+        jButtonLoadFromFile.addActionListener(this);
+        add(jButtonLoadFromFile);
+
+        jLabelSelectedFile = new JLabel(NO_FILE_SELECTED, SwingConstants.CENTER);
+        jLabelSelectedFile.setBounds(20, 40, 300, 30);
+        add(jLabelSelectedFile);
+
+        jButtonLoadScenario = new JButton("Load");
+        jButtonLoadScenario.setBounds(320, 40, 100, 30);
+        jButtonLoadScenario.addActionListener(this);
+        add(jButtonLoadScenario);
+
+
+        jComboBoxScenarios = new JComboBox(possibleScenariosList.toArray());
+        jComboBoxScenarios.setBounds(440, 10, 400, 30);
+        add(jComboBoxScenarios);
+
+
+        jButtonGenerateScenario = new JButton("Generate");
+        jButtonGenerateScenario.setBounds(440, 40, 400, 30);
+        jButtonGenerateScenario.addActionListener(this);
+        add(jButtonGenerateScenario);
+
+        jLabelImageScenario = new JLabel(getStartingImageIcon());
+        jLabelImageScenario.setBounds(20, 80, IMAGE_WIDTH, IMAGE_HEIGHT);
+        add(jLabelImageScenario);
+
+        jLabelSelectPhilosophyPrompt = new JLabel("Select custom philosophy");
+        jLabelSelectPhilosophyPrompt.setBounds(50, 515, 200, 30);
+        add(jLabelSelectPhilosophyPrompt);
+
+        jButtonAddCustomPhilosophy = new JButton("Add new");
+        jButtonAddCustomPhilosophy.setBounds(CENTER_CUSTOM_PHILOSOPHIES + 520, 500, 100, 30);
+        jButtonAddCustomPhilosophy.addActionListener(this);
+        add(jButtonAddCustomPhilosophy);
+
+
+        jComboBoxCustomPhilosophies = new JComboBox(getCustomPhilosophiesNames().toArray());
+        jComboBoxCustomPhilosophies.setBounds(CENTER_CUSTOM_PHILOSOPHIES + 220, 500, 300, 30);
+        add(jComboBoxCustomPhilosophies);
+
+        jButtonCustomPhilosophyInfo = new JButton("Info");
+        jButtonCustomPhilosophyInfo.setBounds(CENTER_CUSTOM_PHILOSOPHIES + 220, 530, 200, 30);
+        jButtonCustomPhilosophyInfo.addActionListener(this);
+        add(jButtonCustomPhilosophyInfo);
+
+        jButtonCalculate = new JButton("Calculate");
+        jButtonCalculate.setBounds(CENTER_CUSTOM_PHILOSOPHIES + 420, 530, 200, 30);
+        jButtonCalculate.addActionListener(this);
+        add(jButtonCalculate);
+
+        jLabelBestDecision = new JLabel("");
+        jLabelBestDecision.setBounds(50, 600, 400, 30);
+        jLabelBestDecision.setVisible(false);
+        add(jLabelBestDecision);
+
+
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        Object eventSource = e.getSource();
+
+        if (eventSource == jButtonLoadFromFile) {
```

Comment:
dla każdego takiego ifa, masz jakąś implementację w środku, może wyciągnij te implementacje do osobnych funkcji, w stylu HandlaLoadFromFileButtonPressed(), łatwiej to się będzie modyfikować w przyszłości